src/plugins/gui/sepaPlugin.ts

```diff
@@ -0,0 +1,347 @@
+              }
+            ]
+          },
+          {
+            title: s.strings.home_address_title,
+            key: 'addressForm',
+            formType: 'addressForm',
+            fields: [
+              // TODO: Localize format based on selected country.
+              // For now, use a "generic Amazon.com" format
+              {
+                key: 'address',
+                label: s.strings.address_line_1,
+                dataType: 'address'
+              },
+              {
+                key: 'address2',
+                label: s.strings.address_line_2,
+                dataType: 'text',
+                optional: true
+              },
+              {
+                key: 'city',
+                label: s.strings.city,
+                dataType: 'text'
+              },
+              {
+                key: 'state',
+                label: s.strings.state_province_region,
+                dataType: 'text'
+              },
+              {
+                key: 'postalCode',
+                label: s.strings.zip_postal_code,
+                dataType: 'text'
+              }
+            ]
+          }
+        ],
+        onSubmit: async (fieldInputs: FormFieldProps[]) => {
+          // Validate user input
+          const rawSepaInfoInput: any = fieldInputs.reduce(
+            (acc: any, { key, value }) => {
+              if (key === 'iban' || key === 'swift') acc[key] = value
+              else acc.ownerAddress[key] = value
+              return acc
+            },
+            {
+              iban: '',
+              swift: '',
+              ownerAddress: {
+                name: '',
+                address: '',
+                address2: '',
+                city: '',
+                country: regionCode.countryCode,
+                state: '',
+                postalCode: ''
+              }
+            }
+          )
+          try {
+            sepaInfo = asFiatSepaInfo(rawSepaInfoInput)
+          } catch (e: any) {
+            const requiredSepaKeys = ['name', 'iban', 'swift', 'address', 'city', 'country', 'state', 'postalCode']
+            throw new Error(
+              `Missing sepa fields: ${requiredSepaKeys.filter(
+                requiredSepaKey => rawSepaInfoInput[requiredSepaKey] == null && rawSepaInfoInput.ownerAddress[requiredSepaKey] == null
+              )}`
+            )
+          }
+
+          // Get quote from user amount input
+          if (sepaInfo == null) return
+
+          const coreWallet = account.currencyWallets[walletId]
+          const fiatCurrencyCode = 'iso:EUR' // TODO: Allow other accepted fiat codes
+          const displayFiatCurrencyCode = fiatCurrencyCode.replace('iso:', '')
+          const currencyPluginId = coreWallet.currencyInfo.pluginId
+          let enterAmountMethods: FiatPluginGetMethodsResponse
+          let counter = 0
+          let bestQuote: FiatProviderQuote | undefined
+
+          await showUi.enterAmount({
```

Comment:
You should start with enterAmount. Always capture a quote first (if possible) before getting user info.